### PR TITLE
Team/hogwartsrejects/unit tests

### PR DESCRIPTION
**Pull request by vivekshresta**
Dependencies that were added:

junit5 jupiter-5.6.2 for mft-core, resource-service/server, secret-service/server
Mockito-core 3.3.3 for mft-core, resource-service/server, secret-service/server
MockMaker mockito plugin for mocking final classes added to source-service/server, secret-service/server
Dependencies for other transports IN TEST SCOPE had to be added for the core module for writing tests for ConnectorResolver and MetadataCollectorResolver

Tests:
TestConnectorResolver for ConnectorResolver.java - 25 test cases
TestFileBasedResourceBackend for FileBasedResourceBackend - 16 test cases
TestFileBasedSecretBackend for FileBasedSecretBackend - 12 test cases
TestMetadataCollectorResolver for MetadataCollectorResolver - 9 test cases
TestResourceServiceHandler for ResourceServiceHandler - 80 test cases
TestSecretServiceHandler for SecretServiceHandler - 52 test cases
TestSQLResourceBackend for SQLResourceBackend - 13 test cases

Total - 207 unit tests

Since none of the classes had any unit tests, a lot of tests had to be written for handling as many edge cases as possible for all the transports.
Tests were written for all the transports except for FTP and GDrive since they're not yet merged to airavata-mft.

I've made sure to refactor code as less as possible.

### CODE REVIEW

File: services/resource-service/server/src/test/java/org/apache/airavata/mft/resource/server/backend/file/TestFileBasedResourceBackend.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+//        FTPResourceGetRequest resourceGetRequest = Mockito.mock(FTPResourceGetRequest.class);
+//        Mockito.when(resourceGetRequest.getResourceId()).thenReturn(resourceId);
+//
+//        try {
+//            Optional<FTPResource> ftpResourceOptional = fileBasedResourceBackend.getFTPResource(resourceGetRequest);
+//            assertTrue(ftpResourceOptional.isPresent());
+//
+//            FTPResource ftpResource = ftpResourceOptional.get();
+//            assertEquals(resourceId, ftpResource.getResourceId());
+//            assertNotNull(ftpResource.getResourcePath());
```
Comment (vivekshresta) [inline services/resource-service/server/src/test/java/org/apache/airavata/mft/resource/server/backend/file/TestFileBasedResourceBackend.java:35]: Since the contents in resources.json might change, instead of asserting on the content, I'm asserting to make sure the property is not null.

**Comment by DImuthuUpe** [comment]
@vivekshresta Fix conflicts and update the PR

**Comment by vivekshresta** [comment]
> @vivekshresta Fix conflicts and update the PR

Hi Dimuthu,

The merge conflicts are not resolved.

**Comment by DImuthuUpe** [comment]
@vivekshresta  Still there are merge conflicts. Rebase again

**Comment by vivekshresta** [comment]
Hi @DImuthuUpe . Thank you for reviewing my code contribution.
I do not see any merge conflicts and just to be sure, I did try rebasing from upstream and it looks like my branch has the latest upstream commit.
Are you by any chance talking about the possible merge conflicts that might occur when the PR https://github.com/apache/airavata-mft/pull/11 goes through?

I've attached my PR view and I do not see any possible merge conflicts. Please let me know if I'm missing something very obvious.
Screen Shot 2020-05-05 at 1 16 27 AM

File: core/pom.xml [hunk 2]
```diff
@@ -33,11 +33,67 @@
...
+        <dependency>
+            <groupId>org.apache.airavata</groupId>
+            <artifactId>mft-scp-transport</artifactId>
+            <version>0.01-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+         <dependency>
+            <groupId>org.apache.airavata</groupId>
+            <artifactId>mft-dropbox-transport</artifactId>
+            <version>0.0.1-SNAPSHOT</version>
```
Comment (vivekshresta) [inline core/pom.xml:75]: The mft-dropbox-transport's jar file version should ideally be 0.01 but since the convention followed is 0.0.1, I used the same here. 
@DImuthuUpe please let me know if you want me to fix this in a separate PR.

Comment (DImuthuUpe) [reply core/pom.xml:75]: Please fix this in a different PR. I can still see merge conflicts. 

Screen Shot 2020-05-08 at 12 01 08 PM

**Comment by vivekshresta** [comment]
Hi @DImuthuUpe , I'm not sure why I'm not able to see the merge conflicts.
But I do see a new issue with my PR, a cyclic dependency. While writing unit tests for ConnectorResolver.java and MetadataCollectorResolver.java, I was getting ClassNotFoundException, so I added the dependency for all the transports in mft-core's pom.xml, test scope.
But now the problem is, all the transport modules have a dependency on mft-core and adding the transport dependencies in mft-core caused a cyclic dependency error.
Finally, I believe that without the new dependencies, the unit tests will not work, but adding the dependencies causes cyclic dependency errors. I believe only integration tests can be written to cover this code, so I'm planning on removing the changes I've made in the mft-core module(for now).
Please let me know your thoughts.
